src/handlers/s3.js

```diff
@@ -1,15 +1,39 @@
 const { AwsClient } = require('aws4fetch');
 const utils = require('../utils');
 
-function s3HandlerFactory({ accessKeyId, secretAccessKey, bucket, region }) {
+function s3HandlerFactory({
+  accessKeyId,
+  secretAccessKey,
+  bucket,
+  region,
+  endpoint,
+  forcePathStyle,
+}) {
   const aws = new AwsClient({
     accessKeyId,
     region,
     secretAccessKey,
   });
 
+  let resolvedEndpoint; // See https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-bucket-intro.html
```

Comment:
Think this looks good. If we were to break this out to a function we could make an early exit instead? So something like:
```
function getEndpoint(...) {
  if (endpoint && forcePathStyle) {
    const url = new URL(endpoint);
    return`${url.protocol}//${url.host}/${bucket}`;
  }
  
  ... other if's
```

Reply:
done